pkg/edp/client.go

```diff
@@ -0,0 +1,109 @@
+package edp
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/pkg/errors"
+
+	"github.com/sirupsen/logrus"
+)
+
+type Client struct {
+	HttpClient *http.Client
+	Config     *Config
+	Logger     *logrus.Logger
+}
+
+const (
+	edpPathFormat   = "%s/namespaces/%s/dataStreams/%s/%s/dataTenants/%s/%s/events"
+	contentType     = "application/json;charset=utf-8"
+	userAgentMetris = "metris"
+)
+
+func NewClient(config *Config, logger *logrus.Logger) *Client {
+	httpClient := &http.Client{
+		Transport: http.DefaultTransport,
+		Timeout:   config.Timeout,
+	}
+	return &Client{
+		HttpClient: httpClient,
+		Logger:     logger,
+		Config:     config,
+	}
+}
+
+func (eClient Client) NewRequest(dataTenant string) (*http.Request, error) {
+	edpURL := fmt.Sprintf(edpPathFormat,
+		eClient.Config.URL,
+		eClient.Config.Namespace,
+		eClient.Config.DataStreamName,
+		eClient.Config.DataStreamVersion,
+		dataTenant,
+		eClient.Config.DataStreamEnv,
+	)
+
+	eClient.Logger.Debugf("sending event to '%s'", edpURL)
+	req, err := http.NewRequest(http.MethodPost, edpURL, bytes.NewBuffer([]byte{}))
+	if err != nil {
+		return nil, fmt.Errorf("failed generate request for EDP, %d: %v", http.StatusBadRequest, err)
+	}
+
+	req.Header.Set("User-Agent", userAgentMetris)
+	req.Header.Add("Content-Type", contentType)
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", eClient.Config.Token))
+
+	return req, nil
+}
+
+func (eClient Client) Send(req *http.Request, payload []byte) (*http.Response, error) {
+	var resp *http.Response
+	var err error
+	// TODO make it configurable
```

Comment:
Is this `TODO` needed for this release or future improvement?

Reply:
I already had the timeout hence using it in duration and removed the TODO